pkg/workflow/workflow.go

```diff
@@ -0,0 +1,94 @@
+package workflow
+
+import "context"
+
+// TaskID is a string that uniquely identifies a task.
+type TaskID string
+
+// Task represents an individual step within a workflow that can be run.
+type Task interface {
+	// ID returns a unique identifier for the task.
+	ID() TaskID
+
+	// RunTask executes the task. Tasks may return a context that should be used in subsequent task
+	// execution.
+	RunTask(context.Context) (context.Context, error)
```

Comment:
Returning a context was necessary for information not available on construction, such as the bootstrap cluster config. 

We could write a custom context that lets us populate the parameter passed instead of treating it as immutable.